src/targets/gpu/kernels/include/migraphx/kernels/roialign.hpp

```diff
@@ -92,15 +92,15 @@ MIGRAPHX_DEVICE_CONSTEXPR typename Iterator::value_type bilinear_interpolate(
                                 high[0] * dims[1] + low[1],
                                 high[0] * dims[1] + high[1]};
 
-    float ly                                   = xy[0] - low[0];
-    float lx                                   = xy[1] - low[1];
-    float hy                                   = 1.0f - ly;
-    float hx                                   = 1.0f - lx;
-    array<typename Iterator::value_type, 4> ws = {hy * hx, hy * lx, ly * hx, ly * lx};
+    float ly           = xy[0] - low[0];
+    float lx           = xy[1] - low[1];
+    float hy           = 1.0f - ly;
+    float hx           = 1.0f - lx;
+    array<float, 4> ws = {hy * hx, hy * lx, ly * hx, ly * lx};
```

Comment:
Why change this from Iterator::value_type to float? Is the expectation that this will get converted down to fp8 easier?

Reply:
hy, hx, lx , ly all are float. Therefore better to do comptuation in float and then convert final result to expected return type.

Reply:
Ah I see you want to preserve precision till the last convert. Got it.